test/lib/notifier_test.rb

```diff
@@ -16,13 +28,64 @@ def test_notifier_with_no_s3_path
     end
   end
 
-  def test_export_date_override
+  def test_exists_export_with_done_file
+    AuroraBootstrapper::Notifier.any_instance.stubs( :client ).returns( @stub_client )
+    assert AuroraBootstrapper::Notifier.new(s3_path: @bukkit).exists_export?(date: DateTime.now-1)
+  end
+
+  # assert_output( /Export succeeded/ ) do
+  #   assert @table.export!( into_bucket: "s3://bukkit" )
+  # end
+  def test_exists_export_with_zero_file
+    stubbed_objs = {
+    }
+    local_stub_client = Aws::S3::Client.new(stub_responses: {
+      list_objects_v2: stubbed_objs,
+    })
+
+    AuroraBootstrapper::Notifier.any_instance.stubs( :client ).returns( local_stub_client )
+    assert !AuroraBootstrapper::Notifier.new(s3_path: @bukkit).exists_export?(date: DateTime.now-1)
+  end
+
+  def test_exists_export_without_done_file
+    stubbed_objs = {
+      contents: [
+        { key: 'other.txt' },
+      ]
+    }
+    local_stub_client = Aws::S3::Client.new(stub_responses: {
+      list_objects_v2: stubbed_objs,
+    })
+
+    AuroraBootstrapper::Notifier.any_instance.stubs( :client ).returns( local_stub_client )
+    assert !AuroraBootstrapper::Notifier.new(s3_path: @bukkit).exists_export?(date: DateTime.now-1)
+  end
+
+  def test_export_date_override_with_done_file
+    AuroraBootstrapper::Notifier.any_instance.stubs( :client ).returns( @stub_client )
+    assert_equal DateTime.now.strftime("%Y-%m-%d"), AuroraBootstrapper::Notifier.new(s3_path: @bukkit).export_date_override
```

Comment:
it is encouraged when testing against `Time.now`-based code to stub time to an expected value. Otherwise you run the risk of intermittently failing tests that run around midnight. Not a real concern here, but worth addressing just as best practice

Reply:
May need some extra work here. Let me think ...

Reply:
In the code it checks last 30 days. Looks like the code logic is always related to NOW

Reply:
Stubbing Date.today